private/buf/bufsync/bufsynctest/existing_remote_branch.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bufsynctest
+
+import (
+	"testing"
+
+	"github.com/bufbuild/buf/private/buf/bufsync"
+	"github.com/bufbuild/buf/private/pkg/git"
+	"github.com/bufbuild/buf/private/pkg/git/gittest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func testExistingRemoteBranchNotPreviouslySyncedContentMatch(t *testing.T, handler TestHandler, run runFunc) {
+	repo := gittest.ScaffoldGitRepository(t)
```

Comment:
Nit: for easier reading, differentiate Git vs BSR entities
```suggestion
	gitRepo := gittest.ScaffoldGitRepository(t)
```